vector/src/main/java/im/vector/app/features/home/room/detail/composer/VoiceMessageHelper.kt

```diff
@@ -76,13 +76,17 @@ class VoiceMessageHelper @Inject constructor(
         startRecordingAmplitudes()
     }
 
-    fun stopRecording(): MultiPickerAudioType? {
+    fun stopRecording(convertForSending: Boolean): MultiPickerAudioType? {
         tryOrNull("Cannot stop media recording amplitude") {
             stopRecordingAmplitudes()
         }
         val voiceMessageFile = tryOrNull("Cannot stop media recorder!") {
             voiceRecorder.stopRecord()
-            voiceRecorder.getVoiceMessageFile()
+            if (convertForSending) {
+                voiceRecorder.getVoiceMessageFile()
+            } else {
+                voiceRecorder.getCurrentRecord()
```

Comment:
I'm wondering if in this case the MultiPickerAudioType that we are creating below will still be OK once the file is actually converted.

Reply:
Or we will create a new one. Let me do some test :)

Reply:
if we continue to use the original uri it'll be invalid as the file will not longer exist, will do some extra testing

Reply:
Tested ok on my API 28 emulator. This fun is called when we send the event so I think this is OK.